traj/src/traj/cubic_eq_roots.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+
+import math
+
+#to find the real root/roots of a cubic equation
+def real_roots_cubic_eq ( a,  b,  c,  d):
+#    print "real_roots_cubic_eq: a, b, c, d= "
+#    print  a,  b, c, d
+    
+    rt1=0
+    rt2=0
+    rt3=0
+    if (a == 0.0000000):
+#        print "The coefficient of the cube of x is 0. Please use the utility for a SECOND degree quadratic. No further action taken."
+        rt1, rt2, n_rts = quad_eq_real_root ( b, c, d)
+        return rt1, rt2, -100, n_rts
+    
+    if ( abs(d) < 1e-10):
+#        print "One root is 0. Now divide through by x and use the utility for a SECOND degree quadratic to solve the resulting equation for the other two roots. No further action taken."
+        rt1 = 0
+        rt2, rt3, n_rts = quad_eq_real_root (a, b, c)
+        return rt1, rt2, rt3, n_rts+1
+    
+    b /= a
+    c /= a
+    d /= a
+#    double disc, q, r, dum1, s, t, term1, r13;
+    q = (3.0*c - (b*b))/9.0
+    r = -(27.0*d) + b*(9.0*c - 2.0*(b*b))
+    r /= 54.0
+    disc = q*q*q + r*r
+#    print "disc= {}".format(disc)
+    term1 = (b/3.0)
+    if (disc > 0.0): # one root real, two are complex
+#        print "disc > 0"
+        s = r + math.sqrt(disc)
+#        s = ((s < 0) ? -pow(-s, (1.0/3.0)) : pow(s, (1.0/3.0)))
```

Comment:
Consider deleting the commented lines. If you have debugging output that you want to keep in the code but not use every time, you can use `rospy.logdebug()`

Reply:
print lines were changed  to 

> rospy.logdebug()